src/dcat_ap_no_validator_service/service/validator_service.py

```diff
@@ -107,25 +123,39 @@ def _expand_objects_triples(self) -> None:
                         except FetchError:
                             logging.debug(traceback.format_exc())
                             pass
+                        except SyntaxError:
+                            logging.debug(traceback.format_exc())
+                            pass
 
-    def _load_ontologies(self) -> None:  # pragma: no cover
-        """Load relevant ontologies into ontology graph."""
-        # TODO: this loop should be parallellized
-        # TODO: discover relevant ontologies dynamically, should be cached
-        # TODO: cover with proper tests
-        ontologies = [
-            "https://www.w3.org/ns/regorg",
-            "https://www.w3.org/ns/org",
-            "https://raw.githubusercontent.com/Informasjonsforvaltning/organization-catalogue/master/src/main/resources/ontology/org-status.ttl",  # noqa
-            "http://publications.europa.eu/resource/authority/licence",
-        ]
-        for o in ontologies:
-            if (o, None, None) not in self.ontology_graph:
-                logging.debug(f"Trying to add remote ontology {o}")
-                try:
-                    g = fetch_graph(o)
-                    if g:
-                        self.ontology_graph += g
-                except FetchError:
-                    logging.debug(traceback.format_exc())
-                    pass
+    def _import_ontologies(self) -> None:
+        """Import relevant ontologies into ontology graph.
+
+        Interpret the owl import statements. Essentially, recursively merge with all the objects in the owl import
+        statement, and remove the corresponding triples from the graph.
+
+        Based on https://owl-rl.readthedocs.io/en/latest/_modules/owlrl.html#interpret_owl_imports
+        """
+        while True:
+            # 1. collect the import statements:
+            all_imports = [
+                t for t in self.ontology_graph.triples((None, OWL.imports, None))
+            ]
+            if len(all_imports) == 0:
+                # no import statement whatsoever, we can go on...
+                return
+            # 2. remove all the import statements from the graph
+            for t in all_imports:
+                self.ontology_graph.remove(t)
+            # 3. get all the imported vocabularies and import them
+            for (_s, _p, uri) in all_imports:
+                if (uri, None, None) not in self.data_graph:
+                    if (uri, None, None) not in self.ontology_graph:
+                        logging.debug(f"Trying to fetch remote triples about {uri}")
+                        try:
+                            g = fetch_graph(uri)
```

Comment:
Er det koronarestriksjoner på lengden av variabelnavn?

Reply:
Nei, men akkurat `g` har nesten blitt standard variabel når det gjelder å holde på instanser av Graph(). Eg kan godt prefixe med `_` for understreke at denne er "privat": `_g`

Reply:
Privatisert variabel i f8028dd